frontend/src/metabase/dashboard/components/DashboardSettingsSidebar/DashboardSettingsSidebar.tsx

```diff
@@ -0,0 +1,124 @@
+import { useDispatch } from "metabase/lib/redux";
+import { PLUGIN_CACHING } from "metabase/plugins";
+import { Switch } from "metabase/ui";
+import type { CacheableDashboard, Dashboard } from "metabase-types/api";
+
+interface DashboardSettingsSidebarProps {
+  dashboard: Dashboard;
+  onClose: () => void;
+}
+
+export function DashboardSettingsSidebar({
+  dashboard,
+  onClose,
+}: DashboardSettingsSidebarProps) {
+  const [page, setPage] = useState<"default" | "caching">("default");
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  useMount(() => {
+    // this component is not rendered until it is "open"
+    // but we want to set isOpen after it mounts to get
+    // pretty animations
+    setIsOpen(true);
+  });
+
+  if (page === "caching") {
+    return (
+      <PLUGIN_CACHING.SidebarCacheForm
+        item={dashboard as CacheableDashboard}
+        model="dashboard"
+        onBack={() => setPage("default")}
+        onClose={onClose}
+        pt="md"
+      />
+    );
+  }
+
+  return (
+    <ErrorBoundary>
+      <Sidesheet
+        isOpen={isOpen}
+        title={t`Dashboard settings`}
+        onClose={onClose}
+        data-testid="dashboard-settings-sidebar"
+      >
+        <DashboardSidesheetBody
+          dashboard={dashboard}
+          page={page}
+          setPage={setPage}
+          isOpen={isOpen}
+          onClose={onClose}
+        />
+      </Sidesheet>
+    </ErrorBoundary>
+  );
+}
+
+export type DashboardSidebarPageProps = {
+  dashboard: Dashboard;
+  page: "default" | "caching";
+  setPage: Dispatch<SetStateAction<"default" | "caching">>;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const DashboardSidesheetBody = ({
+  dashboard,
+  setPage,
+}: DashboardSidebarPageProps) => {
+  const dispatch = useDispatch();
+
+  const handleToggleAutoApplyFilters = useCallback(
+    (isAutoApplyingFilters: boolean) => {
+      dispatch(toggleAutoApplyFilters(isAutoApplyingFilters));
+    },
+    [dispatch],
+  );
+
+  const autoApplyFilterToggleId = useUniqueId();
+  const canWrite = dashboard.can_write && !dashboard.archived;
+
+  const isCacheable = isDashboardCacheable(dashboard);
+  const showCaching = canWrite && PLUGIN_CACHING.isGranularCachingEnabled();
+
+  if (dashboard.archived) {
+    return null;
+  }
+
+  return (
+    <>
+      <SidesheetCard title={t`General`}>
+        <Switch
+          disabled={!canWrite}
+          label={t`Auto-apply filters`}
+          labelPosition="left"
+          variant="stretch"
+          size="sm"
+          id={autoApplyFilterToggleId}
+          checked={dashboard.auto_apply_filters}
+          onChange={e => handleToggleAutoApplyFilters(e.target.checked)}
+        />
+      </SidesheetCard>
+      {showCaching && isCacheable && (
+        <SidesheetCard title={t`Caching`}>
+          <PLUGIN_CACHING.SidebarCacheSection
+            model="dashboard"
+            item={dashboard}
+            setPage={setPage}
+          />
+        </SidesheetCard>
```

Comment:
The SidebarCacheSection has "Caching policy" as a subtitle, but I think it's meant to be "When to show new results" ([figma](https://www.figma.com/design/NOCVfjRaqkVhNqNq8jL9mX/Rework-the-info-sidebar-and-entity-actions?node-id=6074-10147&node-type=frame&t=JGEvh5IPHjGcYWEC-0))

Reply:
![Screen Shot 2024-09-23 at 3 21 20 PM](https://github.com/user-attachments/assets/f9a7cfc8-c2ba-4f7e-93a7-83cec15c03c9)